src/artifacts_poetry_plugin/plugin.py

```diff
@@ -4,23 +4,141 @@
 import poetry.console.application
 from poetry.factory import Factory
 from cleo.commands.command import Command
+import poetry.config.config
+from poetry.publishing.uploader import Uploader
+from poetry.utils.authenticator import Authenticator
+from poetry.core.packages.package import Package
+from cleo.helpers import argument
+from typing import List
+import logging
 
-ARTIFACTS_LIST = "artifacts-list"
+ARTIFACTS_DEPLOY = "artifacts-deploy"
+REPOSITORY_NAME = "repository_name"
+logger = logging.getLogger(__name__)
 
 
-def getPackages(poetry):
+def get_repository_data(config, repo_name, io):
```

Comment:
C: Why are we just adding methods to the root file instead of attaching them to the plugin class? Do they make sense for use outside of that class? Do we expect them to be used outside of that class?